api/v1beta1/slice_types.go

```diff
@@ -65,6 +65,18 @@ type SliceConfig struct {
 	SliceIpam SliceIpamConfig `json:"sliceIpam"`
 	// ExternalGatewayConfig determines istio ingress/egress configuration
 	ExternalGatewayConfig *ExternalGatewayConfig `json:"externalGatewayConfig,omitempty"`
+	// Namespace Isolation profile contains fields related to namespace binding to slice
+	NamespaceIsolationProfile NamespaceIsolationProfile `json:"namespaceIsolationProfile,omitempty"`
+}
```

Comment:
All the optional fields should be pointers (those with `omitempty` flag). Otherewise, if you apply an object without defining these optional fields, they will be populated with golang default values. Also make sure there are nil checks everywhere

Reply:
done